src/views/product-downloads-view/index.tsx

```diff
@@ -69,26 +74,34 @@ const ProductDownloadsViewContent = ({
 		() => initializeBreadcrumbLinks(currentProduct, isEnterpriseMode, pathname),
 		[currentProduct, isEnterpriseMode, pathname]
 	)
+
+	const [installViewNavItems, setInstallViewNavItems] =
+		useState(sidebarMenuItems)
+
 	const sidebarNavDataLevels = [
 		generateTopLevelSidebarNavData(currentProduct.name),
 		generateProductLandingSidebarNavData(currentProduct),
 		generateInstallViewNavItems(
 			currentProduct,
-			sidebarMenuItems,
+			installViewNavItems as MenuItem[],
 			isEnterpriseMode
 		),
 	]
 
+	useEffect(() => {
+		const updatedHeadings = generateTableOfContentsSidebar(
```

Comment:
Can you walk me through why querying the HTML is the chosen approach here? My first instinct is that `product-downloads-view` should have a pretty good idea of what it's rendering, and thus should be able to provide that data to the sidebar without querying the DOM. Approaches like this will prevent us from adopting split server/client rendering when we're adopting App Router, so I want to make sure we understand the reasons we went with this approach.

Reply:
Great question. When the product version is changed, the headings under "Operating Systems" update to reflect that. Examples below:
Latest version:
<img width="1351" alt="Screenshot 2023-12-04 at 4 25 35 PM" src="https://github.com/hashicorp/dev-portal/assets/55773810/d43fb933-7e69-45e5-8adb-f71183ca9e95">

Oldest version:
<img width="1341" alt="Screenshot 2023-12-04 at 4 25 27 PM" src="https://github.com/hashicorp/dev-portal/assets/55773810/ed4be6b3-c186-42af-8279-74b17cac58fd">

Boundary has a [Desktop Client card ](https://dev-portal-git-heat-featinstall-page-anchor-sidebar-hashicorp.vercel.app/boundary/install#Desktop-client) that isn't included in the fetched data for each version's available downloads. Using the document query works with this current implementation, although the [fetch-release-data function](https://github.com/hashicorp/dev-portal/blob/main/src/lib/fetch-release-data.ts#L113) could be refactored.

This implementation also allows headings to be generated from `/<product>/install-landing.json`, which was a requirement for Consul as they requested links to Kubernetes installs; simply spreads the existing array `[...sidebarMenuItems]`.

I think this could be refactored in a future PR (I can create a task in the install Asana board), but this implementation does work for the current use case.